src/features/accessibility.py

```diff
@@ -1,77 +1,86 @@
-                        strategy=strategy,
-                    )
+                    self._execute_api_call(url=site.url, session=session, strategy=strategy)
                     for strategy in [_DESKTOP, _MOBILE]
                 ]
             )
-        # take the average score over desktop and mobile.
-        # fixme: do not mutate website data.
-        website_data.score = (score[0] + score[1]) / 2
-        return [str(website_data.score)]
-
-    def _decide(self, website_data: WebsiteData) -> tuple[StarCase, Explanation]:
-        if score := getattr(website_data, "score", False):
-            if score <= self.star_levels[0]:
-                return StarCase.ZERO, _LIGHTHOUSE_SCORE_TOO_LOW
-            elif score <= self.star_levels[1]:
-                return StarCase.ONE, _LIGHTHOUSE_SCORE_TOO_LOW
-            elif score <= self.star_levels[2]:
-                return StarCase.TWO, _LIGHTHOUSE_SCORE_TOO_LOW
-            elif score <= self.star_levels[3]:
-                return StarCase.THREE, _LIGHTHOUSE_SCORE_TOO_LOW
-            elif score <= self.star_levels[4]:
-                return StarCase.FOUR, _LIGHTHOUSE_SCORE_SUITABLE
-            elif score > self.star_levels[4]:
-                return StarCase.FIVE, _LIGHTHOUSE_SCORE_SUITABLE
+
+            scores = AccessibilityScores(
+                desktop_score=scores[0], mobile_score=scores[1], average_score=(scores[0] + scores[1]) / 2
+            )
+
+            if scores.average_score <= self.star_levels[0]:
+                return StarCase.ZERO, _LIGHTHOUSE_SCORE_TOO_LOW, scores
+            elif scores.average_score <= self.star_levels[1]:
+                return StarCase.ONE, _LIGHTHOUSE_SCORE_TOO_LOW, scores
+            elif scores.average_score <= self.star_levels[2]:
+                return StarCase.TWO, _LIGHTHOUSE_SCORE_TOO_LOW, scores
+            elif scores.average_score <= self.star_levels[3]:
+                return StarCase.THREE, _LIGHTHOUSE_SCORE_TOO_LOW, scores
+            elif scores.average_score <= self.star_levels[4]:
+                return StarCase.FOUR, _LIGHTHOUSE_SCORE_SUITABLE, scores
+            else:  # scores.average_score > self.star_levels[4]:
+                return StarCase.FIVE, _LIGHTHOUSE_SCORE_SUITABLE, scores
+
+    async def _execute_api_call(self, url: str, session: ClientSession, strategy: str = "desktop") -> float:
+        params = {
+            "url": url,
+            "category": _ACCESSIBILITY,
+            "strategy": strategy,
+        }
+        response = await session.get(url=f"{self.url}/{_ACCESSIBILITY}", timeout=self.timeout, json=params)
+
+        if response.status == 200:
+            # expected result looks like {"score": [0.123]}
+            return float(json.loads(await response.text())["score"][0])
         else:
-            return StarCase.ZERO, NO_EXPLANATION
+            raise Exception(f"Request to lighthouse failed with {response.status}: {await response.text()}")
```

Comment:
raising a general Exception is rather unspecific. Looking at https://docs.python.org/3/library/exceptions.html#exception-hierarchy nothing better jumps to my mind, maybe "RuntimeError" or "ValueError"?

Reply:
changed to RuntimeError